src/history/hs_conn.c

```diff
@@ -133,22 +133,29 @@ int
 __wt_hs_open(WT_SESSION_IMPL *session, const char **cfg)
 {
     WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *hs_session;
 
     conn = S2C(session);
 
     /* Read-only and in-memory configurations don't need the history store table. */
     if (F_ISSET(conn, WT_CONN_IN_MEMORY | WT_CONN_READONLY))
         return (0);
 
+    /* We need a real session to open the HS file. */
+    WT_RET(__wt_open_internal_session(conn, "hs-open", false, 0, 0, &hs_session));
```

Comment:
One suggestion. Instead of having `WT_RET` here, could init `hs_session` to NULL at the start, use `WT_ERR` here and in the `err` label:
```
err:
    if (hs_session != NULL)
        WT_TRET(__wt_session_close_internal(hs_session));
    return (ret);
```

Reply:
Fixed as suggested.